### PR TITLE
usage caveats: Is complate Right for Me?

### PR DESCRIPTION
**Pull request by FND**
as discussed:

> we should provide nuanced, non-biased (FCVO) guidance to prevent people from choosing complate without being aware of the consequences

in the process, `s/adapter/adaptor/`, because FND ¯\\\_(ツ)_/¯

### CODE REVIEW

File: content/faq.md [hunk 1]
```diff
@@ -1,9 +1,29 @@
-title: complate Frequently Asked Questions 
+title: complate Frequently Asked Questions
 h1: FAQ
-subtitle: Frequently asked questions
+subtitle: Frequently Asked Questions
```
Comment (FND) [inline content/faq.md:3]: see #21 🤷‍♂

File: content/faq.md [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
 safe: false
 
 
+Is complate Right for Me? <span id="trade-offs"></span>
```
Comment (FND) [inline content/faq.md:7]: This is a bit of a tacky title - sort of on purpose, but if anyone has any alternative suggestions, I'm all ears.

File: content/faq.md [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+-------------------------
+
+While complate has many advantages, it also introduces additional complexity:
+The use of JavaScript typically requires both a compiler (for module resolution and to transform JSX) and
+a JS runtime environment (typically provided via [the respective adaptor](getting-started.html)) --
```
Comment (FND) [inline content/faq.md:12]: That's a forward reference, as Getting Started doesn't provide such guidance yet.

File: content/faq.md [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
+see [How It Works](how-it-works.html) for a detailed explanation of that interplay.
```
Comment (FND) [inline content/faq.md:13]: This, too, is a forward reference: I wanna explain how complate works as a view layer, how it expects compiler tooling etc. - based in part on diagrams like this:

complate

File: content/faq.md [hunk 5]
```diff
@@ -28,4 +48,9 @@ HTTP and HTML allow [streaming pieces of content as soon as possible](https://me
 Sometimes also known as "isomorphic JavaScript", this means using the same components (macros) on both the server and the client.
 
 </dd>
-</dl>
\ No newline at end of file
+</dl>
+
+
+[authors will need a basic understanding of JavaScript]: https://medium.com/@Heydon/reluctant-gatekeeping-the-problem-with-full-stack-e9ad836570f6
```
Comment (FND) [inline content/faq.md:54]: I'd rather avoid Medium, but https://www.heydonworks.com/article/reluctant-gatekeeping-the-problem-with-full-stack is [currently inaccessible](https://twitter.com/heydonworks/status/1216753746979557378). 😞

File: content/faq.md [hunk 6]
```diff
@@ -1,9 +1,29 @@
...
+It also means [authors will need a basic understanding of JavaScript], including for debugging.
+
+In our experience, all that pays off when
+native templating mechanisms do not provide sufficient means for abstraction and composition (i.e. components).
+It's also very helpful, even essential, in polyglot environments where the same markup definitions (i.e. macros) should be reused across a variety of technology stacks,
+e.g. due to a stand-alone pattern library and/or [self-contained systems].
+
+Otherwise a simpler, more traditional approach might be warranted,
+using a native templating engine within the respective environment.
+In a Node.js environment, that might include [server-side rendering] with JavaScript frameworks like React, Svelte etc.
```
Comment (FND) [inline content/faq.md:23]:
```suggestion
With Node.js, that might include [server-side rendering] with JavaScript frameworks like React, Svelte etc.
```

(less repetitive)

**Comment by joyheron** [comment]
@FND Do you think it would be at all possible for you to write up the `how-it-works` page this week, even if it is just in bullet points? I think your diagram actually explains a lot of the technicalities, so if we had a page with that diagram and a few sentences (or paragraphs) to explain it, I think it would be sufficient.

**Comment by joyheron** [comment]
I ask because I think this "how-it-works" or "what-is-an-adapter" page is the only crucial element left for the website and I would like to get at least the minimal explanation on the site so that we can call it "good enough" for now.

**Comment by FND** [comment]
Yeah, it's still on my list (though competing with _even more_ items since Thursday... ), and I'd very much like to get it off of there. 🤞

**Comment by FND** [comment]
I went ahead and merged this; we can always improve it later on. (Yeah, right... )

**Pull request merged**